lib/Push.php

```diff
@@ -187,11 +187,18 @@ protected function encryptAndSign(Key $userKey, array $device, int $id, INotific
 		$data = [
 			'nid' => $id,
 			'app' => $notification->getApp(),
-			'subject' => $notification->getParsedSubject(),
+			'subject' => '',
 			'type' => $notification->getObjectType(),
 			'id' => $notification->getObjectId(),
 		];
 
+		// Max length of encryption is 255, so we need to shorten the subject to be shorter
+		$subject = $notification->getParsedSubject();
+		$dataLength = 245 - strlen(json_encode($data));
+		if (strlen($subject) > $dataLength) {
+			$data['subject'] = substr($subject, 0, $dataLength) . '…';
```

Comment:
Should the substr not be 1 shorter than the dataLength?

Reply:
I started with 245, instead of 255 so we have a little buffer there

Reply:
Ah ok perfect. I wasn't sure if that was maybe for other metadata somewhere.